datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -104,6 +113,47 @@ def compute_derivative_rows(self, rows, metrics, key):
         return result
 
 
+def merge_duplicate_rows(rows, metrics, key):
+    """
+    Given a list of query rows, apply query normalization and compute query signatures to detect duplicate
+    queries_by_key and merge them into a single row.
+
+    - **rows** (_List[dict]_) - rows from current check run
+    - **metrics** (_List[str]_) - the metrics to compute for each row
+    - **key** (_callable_) - function for an ID which uniquely identifies a query row across runs
+    """
+
+    queries_by_key = OrderedDict()
+    for row in rows:
+        merged_row = dict(row)
+        try:
+            if 'query' not in row:
+                logger.warning('row missing \'query\' key')
+                continue
+
+            normalized_query = datadog_agent.obfuscate_sql(row['query'])
+            if not normalized_query:
+                logger.warning("Obfuscation of query '%s' resulted in empty query", row['query'])
+                continue
+        except Exception as e:
+            logger.warning("Failed to obfuscate query '%s': %s", row['query'], e)
+            continue
+
+        merged_row['normalized_query'] = normalized_query
```

Comment:
I was adding those entries in the merge row to avoid postgres and mysql needing to recompute the normalized query and query signature but it does feel like this might appear as a surprising side-effect here (highlighted by the existing tests having broken and needing to be updated). 

A few options: 
* Adding a note about this in the `merge_duplicate_rows` and `compute_derivative_rows` doc.
* Removing the extra added entries and just using the normalized query and signature here for merging. This means we'd recompute those in the postgres and mysql integrations to set them in tags. 
* Rename `merge_duplicate_rows` to `normalize_query_rows` (which, I think, opens it up to a more natural augmentation of the row with normalization-related keys along with the merging logic) and have the integrations call `normalize_query_rows` followed by `compute_derivative_rows`.

Reply:
Personally, I would prefer this function remains as simple as possible and does not have any side effects on the data contents of the row. The purpose of the `key` function is to keep the functionality generic across integrations which may have a "query" column (postgres) or a "digest_text" column (mysql) for the SQL text. So the merging should only happen based on the `key` and it would be up to the caller to define a key that uses the query_signature.

To your point, we should really only call `datadog_agent.obfuscate_sql` once per row per integration. So I think a better flow would be one that:

1. The mysql/postgres integration "enriches" each row with the `query_signature`.
2. The mysql/postgres integration calls `compute_derivative_rows` defining their respective `key` functions which includes the `query_signature`.
3. `compute_derivative_rows` calls `merge_duplicated_rows` before returning the derivatives.

Reply:
One other note: the "query" and "digest_text" columns can already have several KB of data per row (or more because python is extremely memory inefficient for storing strings), so doubling it by adding a "normalized_query" key to each row seems unnecessary. The mysql/postgres integrations should probably just replace their respective sql columns with the text of the normalized query since we don't use them after computing the query_signature.

Reply:
> The mysql/postgres integrations should probably just replace their respective sql columns with the text of the normalized query since we don't use them after computing the query_signature.

Yes this is what I'm doing already in the upcoming changes for sending metrics via events https://github.com/DataDog/integrations-core/pull/9223/files#diff-17584d7d2fd6b787e9e92a852f881f84c5dee3af11aa571549593a0a4d2b55e2R116

Reply:
This is great. I think I've got all the answers I was looking for to wrap up this PR for the base changes. Thanks!